### ISSUE TITLE
Generic annotation for @Given @When @Then @And

### ISSUE THREAD

**Issue by ntthaibk** [cucumber/cucumber-jvm#2705]
### 🤔 What's the problem you're trying to solve?

When I create the journey scenario. I use the `*` syntax in .feature file instead of normal Given When Then And. But then in those step's step definition, the @Given or @When @Then @And annotation is not able to differentiate the Journey step and normal integration test steps. 

### ✨ What's your proposed solution?

I think another alias annotation for @StepDefinitionAnnotation is gonna be a good solution, it could be named either `@Steps` or `@Journey`

### 📚 Any additional context?

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/feature_request.md)*

**Comment by mpkorstanje** [comment]
I'd be happy to take a pull request that adds this.

To do this it would probably be best to change the [GenerateI18n.java](https://github.com/cucumber/cucumber-jvm/blob/main/cucumber-java/src/codegen/java/GenerateI18n.java) so that in addition to generating a localized annotation for every `io.cucumber.java.<lang>` it also generates a `@Step` annotation in the `io.cucumber.java` package.

**Comment by ntthaibk** [comment]
hi @mpkorstanje, so I checked the [Generatel18n.java](https://github.com/cucumber/cucumber-jvm/blob/main/cucumber-java/src/codegen/java/GenerateI18n.java) today, and found out that actually I need to modify the [Gherkin Dialect](https://github.com/cucumber/gherkin-java/blob/main/src/main/groovy/io/cucumber/gherkin/GherkinDialects.gsp) and I saw the repo already archived, can I send the MR for the archived repo there, thanks

**Comment by mpkorstanje** [comment]
You can find the code for Gherkin in [cucumber/gherkin](https://github.com/cucumber/gherkin). What made you believe it was [cucumber/gherkin-java](https://github.com/cucumber/gherkin-java)?

**Comment by ntthaibk** [comment]
ahh I see, I thought it was gherkin-java after a google search with the keyword GherkinDialect.java

**Comment by mpkorstanje** [comment]
Btw,  updating `languages.json` is not the right way to go. It will also make `Step` a valid keyword to use in Gherkin.

I don't think you'd have to change `languages.json` anyway . There already is the `*` keyword. However isn't rendered as an annotation because it is an invalid class name and filtered out. You could replace it with something that is valid.



https://github.com/cucumber/cucumber-jvm/blob/12c5029592684c3338f98f656b7de90f81cd6f36/cucumber-java/src/codegen/java/GenerateI18n.java#L76-L80

**Comment by missingfaktor** [comment]
Hi all, is there any update on this issue?

**Comment by mpkorstanje** [comment]
No. Would you be interested in providing a pull requests that adds this feature?
